packages/edit-site/src/components/dataviews/dataviews.js

```diff
@@ -158,6 +159,25 @@ export default function DataViews( {
 			<VStack spacing={ 4 }>
 				<HStack justify="space-between">
 					<TextFilter onChange={ dataView.setGlobalFilter } />
+					{ Object.keys( view.filters ).map( ( key ) => {
+						const field = dataView
+							.getAllColumns()
+							.find(
+								( column ) =>
+									column.columnDef.renderFilter &&
+									column.id === key
+							);
+						if ( ! field ) {
+							return null;
+						}
+
+						return field.columnDef.renderFilter();
```

Comment:
allowing the definition to render the whole filter is probably going to be the fallback. Most filters will use a declarative approach and be rendered within the data view to absorb the complexity.

I guess that's the same as this comment https://github.com/WordPress/gutenberg/pull/55270/files#r1355256193

I'd love to see a couple filters added, which would force us to think about this. Are you thinking this should be a follow-up and focus on the design on the current one?